pandas/core/internals/array_manager.py

```diff
@@ -65,14 +76,80 @@
     Index,
     ensure_index,
 )
-from pandas.core.internals.base import DataManager
+from pandas.core.internals.base import (
+    DataManager,
+    SingleManager,
+)
 from pandas.core.internals.blocks import make_block
 
-if TYPE_CHECKING:
-    from pandas.core.internals.managers import SingleBlockManager
+T = TypeVar("T", bound="ArrayManager")
+
 
+def astype_array(values, dtype, copy):
```

Comment:
This added code here (this function and the one below) is only temporary until https://github.com/pandas-dev/pandas/pull/40141 is merged (that PR is moving the Block.astype implementation so it can be reused here)